src/main/java/uk/gov/pay/products/service/ProductFinder.java

```diff
@@ -22,4 +24,17 @@ public ProductFinder(ProductDao productDao, LinksDecorator linksDecorator) {
         return productDao.findByExternalId(externalId)
                 .map(productEntity -> linksDecorator.decorate(productEntity.toProduct()));
     }
+
+    @Transactional
+    public Optional<Product> disableProduct(String externalId) {
+        Optional<ProductEntity> productEntity = productDao.findByExternalId(externalId);
+        if(productEntity.isPresent()){
```

Comment:
sorry to come back on this again, but I think ideally this would follow our usual pattern of mapping over the optional returned by `findByExternalId`. It is good to have conventions 😄